src/renderer/services/file-management-system/index.ts

```diff
@@ -447,25 +442,31 @@ export default class FileManagementSystem {
         // If FSS is still available to continue receiving chunks of this upload
         // simply continue sending the chunks
         let lastChunkNumber = fssStatus.chunkStatuses.findIndex(
-          (status) => status !== ChunkStatus.COMPLETE
+          (status) => status !== UploadStatus.COMPLETE
         );
         if (lastChunkNumber === -1) {
           lastChunkNumber = fssStatus.chunkStatuses.length;
         }
 
-        const partiallyCalculatedMd5 = upload.serviceFields.md5CalculationInformation?.[`${lastChunkNumber}`];
-        if (!partiallyCalculatedMd5){
-          throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
+        let partiallyCalculatedMd5 = undefined;
+        if (lastChunkNumber > 0) {
+          const chunkResponse = await this.fss.getChunkInfo(fssUploadId, lastChunkNumber);
+          partiallyCalculatedMd5 = chunkResponse.cumulativeMD5;
+          if (!partiallyCalculatedMd5){
+            throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
+          }
         }
 
-        if (fssStatus.status === UploadStatus.RETRY ) {
+        if (fssStatus.status === UploadStatus.RETRY) {
```

Comment:
Does this condition need to be concerned about `partiallyCalculatedMd5 ` ?  If not It looks like this condition could come first, and that would simplify the routine.

Reply:
Yeah it uses it in the retryFinalize call, is that unnecessary to give to FSS on retry?

Reply:
Yeah `retryFinalize` doesnt take and MD5.

But also remember, after https://github.com/aics-int/file-storage-service/pull/210 , RETRY will just mean for the client to call `/finalize` again.  When called in a retry scenerio, the MD5 field can be blank.  If it is not blank, it must have the same value as when `/finalize` was called the first time.